highfidelity/hifi
# Materials on primitives

- adds UVs and tangents to ShapeEntities so they fully support materials
- fixes a normal mapping bug
- fixes https://highfidelity.fogbugz.com/f/cases/12745/Setting-Shape-entity-item-s-alpha-property-to-non-1-makes-entity-render-black

Test plan:
- run [this](https://gist.githubusercontent.com/SamGondelman/3f2db7655b51ae2ffe3741778faef7e2/raw/1cd047ff7cd871a03dc5910c4b35a71622dadb7e/PrimitiveMaterials.js).  You should see a row of 13 shapes and two more on a higher row.
- The bottom row of shapes should all be textured and have a normal map.
- For the first 5 shapes, the black edges in the textures should perfectly line up with their edges.  The numbers around each point should all be the same on the different faces that share that point.
- For the next 6 shapes (up to the flat shapes), you should see the texture wrapped around them once horizontally.  The top and bottom (of the shapes that have them) should look like cut-outs of the texture.
- The cube on the top row should be red and transparent.
- The sphere on the top row should be identical to the sphere below it (it's actually a material entity).
- Go to a domain with procedural shape entities (eschatology is good).  Verify that they still appear correctly.

For future reference, this is how our shapes are UV-mapped (top left is (0, 0), bottom right is (1, 1)):
![texturetetrahedron](https://user-images.githubusercontent.com/7650116/37120321-e9d16d94-220e-11e8-8f0b-b2ecb77a4374.png)
![texturecube](https://user-images.githubusercontent.com/7650116/37120317-e96b63dc-220e-11e8-93e8-64f7425e9932.png)
![textureoctahedron](https://user-images.githubusercontent.com/7650116/37120320-e9b81790-220e-11e8-9f4d-ed73e6617044.png)
![textureicosahedron](https://user-images.githubusercontent.com/7650116/37120319-e9a31020-220e-11e8-95e6-d94b6fcbddc5.png)
![texturedodecahedron](https://user-images.githubusercontent.com/7650116/37120318-e985b020-220e-11e8-93cb-bbee0f01af1e.png)
